sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcUtils.scala

```diff
@@ -372,12 +385,14 @@ object OrcUtils extends Logging {
       canPruneCols: Boolean,
       dataSchema: StructType,
       resultSchema: StructType,
```

Comment:
I'm thinking about this again. It's a long-standing issue that the schema evolution is completely defined by the ORC library, but it's non-trivial to fix this and let Spark take control.

So my proposal is: when we generate `resultSchemaString` and set it to the ORC reader, we should replace `TimestmapNTZType` with `LongType` in the catalyst schema, as we know NTZ is stored as int64 in ORC file.